### DISCUSSION TITLE
Utilità dei plugin

### DISCUSSION THREAD

**Discussion by xrivo1** [rejetto/hfs#214] [Ideas]
Molti plugin, sono utili in quanto le loro funzioni sono dedicate all'effettivo utilizzo che ognuno di noi fa del server, 
Chi condivide foto utilizza il plugin delle anteprime immagini, chi condivide video o audio utilizza il plugin dei lettori multimediali.
Ritengo invece che , quei plugin che rendano il fileserver più sicuro, debbano essere integrati direttamente nel programma. ( vedi anti dos, limitazione di banda, limite accessi ecc.
Non conosco l'impatto sul programma del caricamento dei plugin, ma credo che funzioni integrate snelliscano il tutto.
Cosa ne pensi?

**Comment by rejetto** [comment]
la risposta breve è: sì.
Ora viene la risposta lunga 🫠
Alcuni dei plugin sono nati già con l'idea che sarebbero stati (potenzialmente) integrati nel programma stesso.
Può sembrare strano, ma ci sono diversi motivi per agire così:
- dimostrare le potenzialità dei plugin (non credere che un plugin possa fare tutto)
- mostrare come si fanno
- testarne (io) il buon funzionamento
- permettere/stimolare il potenziamento di una feature (com'è successo con il player, hanno iniziato solo dopo che ho fatto il primo)
- capire come migliorare l'API dei plugin in base a necessità reali

Decidere di "assimilare" un plugin è meno banale di come possa sembrare. Per un buon design devo anche considerare di mantenere basso il numero di opzioni (troppe opzioni = non le guarda nessuno), e mantenere relativamente bassa la complessità del programma (manutenibilità, rischi di sicurezza).

**Reply by xrivo1** [reply]
Ottimo. Grazie
Effettivamente ul plugin è una funzione che puoi abilitare e disabilitare quando vuoi, decidere o meno di installarla e soprattutto poterla modificare/implementare a tuo piacimento creando magari qualcosa di migliore.
